CHANGELOG.md

```diff
@@ -7,6 +7,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ## [12.0.1]
+### Changed
+- Limit the transaction history in the user's data, and in state logs, to 100 entries ([#26236](https://github.com/MetaMask/metamask-extension/pull/26236))
```

Comment:
I would describe this more as a fix, to prevent pending transactions from leading to excessive disk space usage.

The history affected here isn't really user-visible. The minimal use of the history in the activity log is hoped to be mostly unaffected by this change (if it is affected, it would be in a rare edge case). Users would probably think "transaction history" means the list of previous transactions, which is not affected by this change.